server/src/test/java/com/objectcomputing/checkins/services/teammembers/TeamMemberControllerTest.java

```diff
@@ -0,0 +1,237 @@
+import static org.mockito.Mockito.when;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import javax.inject.Inject;
+
+import com.objectcomputing.checkins.services.memberprofile.MemberProfile;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileController;
+
+import com.objectcomputing.checkins.services.team.Team;
+import com.objectcomputing.checkins.services.team.TeamController;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+import io.micronaut.core.type.Argument;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import io.micronaut.test.annotation.MicronautTest;
+
+@TestInstance(Lifecycle.PER_CLASS)
+@MicronautTest
+public class TeamMemberControllerTest {
+
+    @Inject
+    @Client("/services/team-member")
+    private HttpClient client;
+
+    @Inject
+    MemberProfileController memberProfileController;
+
+    @Inject
+    TeamController teamController;
+
+    TeamMemberRepository mockTeamMemberRepository = mock(TeamMemberRepository.class);
+    TeamMember mockTeamMember = mock(TeamMember.class);
+
+    private static UUID testId;
+    private static UUID testTeamMemberId;
+    private static UUID testTeamId;
+    private static boolean isLead = false;
+    private static boolean isDataSetupForTest = false;
+
+    @BeforeAll
+    void setupMemberProfileRecord() {
+    
+        // setup a record in Member-Profile to satisfy foreign key constraint
+        if(memberProfileController != null) {
+            MemberProfile testMemberProfile = new MemberProfile("TestName", 
+                                                                "TestRole", 
+                                                                UUID.randomUUID(), 
+                                                                "TestLocation", 
+                                                                "TestEmail", 
+                                                                "TestInsperityId", 
+                                                                LocalDate.now(), 
+                                                                "TestBio");
+
+            final HttpResponse<?> response = memberProfileController.save(testMemberProfile);
+            assertEquals(HttpStatus.CREATED, response.getStatus());
+            assertNotNull(response.body());
+            testTeamMemberId = ((MemberProfile) response.body()).getUuid();
+        }
+    
+        // setup a record in Member-Profile to satisfy foreign key constraint
```

Comment:
This should be Team and not Member-Profile =)

Reply:
Updated.